tt_metal/impl/device/device.cpp

```diff
@@ -402,13 +401,37 @@ void Device::build_firmware() {
     log_debug(tt::LogMetal, "Building base firmware for device {}", this->id_);
     ZoneScoped;
 
-    this->generate_device_headers(this->build_env_.get_out_firmware_root_path());
+    this->generate_device_bank_to_noc_tables();
     jit_build_set(this->firmware_build_states_, nullptr);
 }
 
+void Device::initialize_device_bank_to_noc_tables(const HalProgrammableCoreType &core_type, CoreCoord phys_core)
+{
+    const uint32_t num_dram_to_noc_words = dram_bank_to_noc_xy_.size() * sizeof(uint16_t);
```

Comment:
are these sizes correct?  size is bytes, sizeof uint16_t is 2.  but you want words, so wouldn't that be size / sizeof(uint32_t)?